packages/firestore/src/api/database.ts

```diff
@@ -552,6 +553,61 @@ export class WriteBatch
   }
 }
 
+class UntypedFirestoreDataConverterAdapter<U>
```

Comment:
Also the obligatory comment on naming: All classes in this file are essentially adapters. I would just use the public API name for this, which follows the usage across this file (e.g. DocumentReference, DocumentSnapshot and the newly introduced FirestoreDataConverter).

Reply:
Done.